app/models/katello/upstream_consumer.rb

```diff
@@ -0,0 +1,61 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class UpstreamConsumer < OpenStruct
+    CP_CONSUMER = Resources::Candlepin::UpstreamConsumer
+
+    class << self
+      def bind_entitlements(organization, params)
+        CP_CONSUMER.organization = organization
```

Comment:
This may have come from your previous PR and if i missed it i apologize (i can't seem to locate where its coming from), but storing values in a class like this is a bad smell and screams that you really want some sort of object instantiated.   If we're not going to pass the org into any method that needs it at the resource layer, we need to refactor some of this to be objects we instantiate and can hold those values).   

In addition this is not thread safe, and may cause issues later on

Reply:
Yeah.... I was actually kind of shocked this didn't come up sooner. It's a major flaw with the candlepin resource code - it's all classes and no objects!

Reply:
Its not great, but not a huge problem untill you start trying to store data like you're doing.  Previously it was stateless, so it it didn't really matter too much.  But you made it stateful :)

Reply:
Can we keep Resources::Candlepin::UpstreamConsumer stateless and just feed it all of the necessary credentials ? ie decouple the current auth stuff embedded in that class?

Reply:
And i don't like the idea of an api layer being stateful with regard to anything other than connection information.  Runcible is a good model IMHO.  You instantiate a runcible object with connection information and options, and then pass in parameter to each call.  It is not stateful with regard to the data that is being passed back and forth.    even if we refactor all of this code to be instantiated objects, I would not advocate having you store the organization on the object itself.  Thats fine for the model layer, but the api layer should be as close to the api as possible IMO

Reply:
It had state before I came along! It just came from settings :) https://github.com/Katello/katello/blob/master/app/lib/katello/resources/candlepin.rb#L53